### PR TITLE
ci: `sdk` and `sdk-audit` in GitHub actions

### PR DESCRIPTION
**Pull request by fionnachan**
audit job requires new advisory from #173 to pass

### CODE REVIEW

File: .github/workflows/build-and-lint.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          node-version: 14
+          cache: 'yarn'
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Lint sdk
+        run: |
+          yarn build --reporter-options output=$TEST_PATH/sdk.xml
+          yarn lint --format junit -o $TEST_PATH/sdk-lint.xml
+      - name: Upload Coverage to Codecov
```
Comment (yahgwai) [inline .github/workflows/build-and-lint.yml:25]: I'm not sure if this is working properly, I see an warning when looking here: https://github.com/OffchainLabs/arbitrum-sdk/runs/8245403463?check_suite_focus=true

`Warning: Codecov: Failed to properly upload: The cwd: ./packages does not exist!`

**Review by yahgwai — COMMENTED** [review]
We also need to remove the old circleci files

File: .github/workflows/build-and-lint.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: 'yarn'
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Lint sdk
+        run: |
+          yarn build --reporter-options output=$TEST_PATH/sdk.xml
+          yarn lint --format junit -o $TEST_PATH/sdk-lint.xml
```
Comment (yahgwai) [inline .github/workflows/build-and-lint.yml:24]: Do you know how we can access these artifacts in the github UI?

Comment (fionnachan) [reply .github/workflows/build-and-lint.yml:24]: I've added a package to upload the artifacts. You can download at the bottom of the summary page:
https://github.com/OffchainLabs/arbitrum-sdk/actions/runs/3014985047

Comment (fionnachan) [reply .github/workflows/build-and-lint.yml:24]: Also added a test report page:
https://github.com/OffchainLabs/arbitrum-sdk/runs/8248976003

File: .github/workflows/build-and-lint.yml (hunk 1 above)
Comment (fionnachan) [reply .github/workflows/build-and-lint.yml:36]: I removed the `work-directory` and it looks similar to that on circleci now.

Both said "No coverage report found. / No coverage files located" though. Should we remove this step?

File: .github/workflows/build-and-lint.yml (hunk 2 above)
Comment (yahgwai) [reply .github/workflows/build-and-lint.yml:24]: nice!

File: .github/workflows/build-and-lint.yml (hunk 1 above)
Comment (yahgwai) [reply .github/workflows/build-and-lint.yml:36]: Hmm, yeh, looks like this hasnt been working for a while - if ever. Lets address it in another PR.

**Pull request merged**